uaclient/contract.py

```diff
@@ -68,30 +70,41 @@ def request_root_macaroon(self):
         self.cfg.write_cache('root-macaroon', root_macaroon)
         return root_macaroon
 
-    def request_accounts(self, macaroon_token):
+    def request_accounts(self, macaroon_token=None, machine_token=None):
         """Request list of accounts this user has access to."""
         headers = self.headers()
-        headers.update({'Authorization': 'Macaroon %s' % macaroon_token})
+        if macaroon_token:
+            headers.update({'Authorization': 'Macaroon %s' % macaroon_token})
+        else:
+            headers.update({'Authorization': 'Bearer %s' % machine_token})
```

Comment:
If we aren't passed either `macaroon_token` or `machine_token`, we will add `Authorization: Bearer None` as a header; we should probably, instead, raise an error (rather than getting an inevitable error after a round-trip to the server).

Reply:
+1 OddBloke, sorry on this branch it was very WorkInProgress to give us something to talk about with ua-conracts folks (I should have indicated as such on the description). The contracts team went a significantly different direction per https://github.com/CanonicalLtd/ua-contracts/pull/238  which involves a big overhaul here. I'm working that now and will ping when this needs re-review.

Reply:
ok think I've addressed most comments

Reply:
My comment here doesn't appear to have been addressed. (I've also unresolved the related comments below.)

Reply:
My fault. I thought I backed out the Bearer Auth optional param when ua-contracts went a different direction. dropped content related to this change from account and contracts request methods.